R/get_delay_estimate.R

```diff
@@ -0,0 +1,80 @@
+#' Estimate a delay distribution from a reporting triangle
+#' @description
+#' Provides an estimate of the reporting delay as a function
+#' of the delay, based on the reporting triangle and the specified maximum
+#' delay and number of reference date observations to be used in the estimation.
+#' This point estimate of the delay is computed empirically, using an
+#' iterative algorithm starting from the most recent observations. It was
+#' modified from the code originally developed by the Karlsruhe Institute
+#' of Technology RESPINOW German Hospitalization Nowcasting Hub,
+#' Modified from: https://github.com/KITmetricslab/RESPINOW-Hub/blob/7cce3ae2728116e8c8cc0e4ab29074462c24650e/code/baseline/functions.R#L55 #nolint
+#' @param triangle matrix of the reporting triangle, with rows representing
+#' the time points of reference and columns representing the delays
+#' indexed at 0
+#' @param max_delay integer indicating the maximum delay to estimate, in units
+#' of the delay. The default is to use the whole reporting triangle,
+#'  `ncol(triangle) -1`.
+#' @param n_history integer indicating the number of reference dates to be
+#' used in the estimate of the reporting delay, always starting from the most
+#' recent reporting delay. The default is to use the whole reporting triangle,
+#' so `nrow(triangle)-1`
+#' @returns delay_df dataframe of length `max_delay` with columns `delay`
+#' and `pmf`, indicating the point estimate of the empirical probability
+#' mass on each delay
+#' @export
+#' @examples
+#' library(epinowcast)
```

Comment:
I'm not sure that using a suggests in the examples without some kind of check will work on say CRAN

Reply:
Probably should consider providing package data with a reporting triangle/a long tidy formatted dataframe so that there's no dependency on epinowcast in the examples

Reply:
Can make a separate issue